### PR TITLE
Docs math proofs

### PR DESCRIPTION
**Pull request by alimalim77**
## Type of Change
- Made Changes to [Past Reviewed Pull Request](https://github.com/Codecademy/docs/pull/2581)
- Created a new concept entry for Discrete Math Proofs
- Added a new subject to subjects.md -> Discrete Math
- Closing Issue -> https://github.com/Codecademy/docs/issues/2391

## Checklist
- [✓] All writings are my own.
- [✓ ] My entry follows the Codecademy Docs style guide.
- [✓] My changes generate no new warnings.
- [✓] I have performed a self-review of my own writing and code.
- [✓] I have checked my entry and corrected any misspellings.
- [✓] I have made corresponding changes to the documentation if needed.
- [✓] I have confirmed my changes are not being pushed from my forked main branch.
- [✓] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [✓] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes # in the "Description" above).

### CODE REVIEW

**Comment by alimalim77** [comment]
@SSwiniarski  Please review this. This is a follow up fresh PR to the suggested changes on PR: #2581 
- Removed yarn file
- Corrected spelling mistakes

File: content/discrete-math/concepts/proofs/proofs.md
Comment (CBID2) [inline content/discrete-math/concepts/proofs/proofs.md:21]: [Typo]
```suggestion
Conjectures are used to construct mathematical proofs. They can be thought of as a building block of a hypothesis that can be validated using logic and patterns.
```

File: content/discrete-math/concepts/proofs/proofs.md
Comment (CBID2) [inline content/discrete-math/concepts/proofs/proofs.md:17]: [Typos]
```suggestion
**Proofs** are a concept in Discrete Mathematics used to verify the truthfulness and validity of statements. They use logical reasoning to establish the certainty of mathematical claims, playing a crucial role in advancing the field of mathematics. Proofs are also helpful in solving the problem partially or completely.
```

File: content/discrete-math/concepts/proofs/proofs.md
Comment (CBID2) [inline content/discrete-math/concepts/proofs/proofs.md:27]: [Typo]

```suggestion
Base case is a term that is described as the initial or smallest value that verifies a statement to be correct. According to the Principle of Mathematical Induction, a proof is bifurcated into two steps:
```

File: content/discrete-math/discrete-math.md
Comment (CBID2) [inline content/discrete-math/discrete-math.md:9]: [Typo]
```suggestion
**Discrete mathematics** is a fundamental branch of mathematics that provides the foundation for various concepts and techniques used in computer science and problem-solving. It encompasses a range of topics, like sets, propositions, relations, functions, logic, proof, and number theory.
```

File: content/discrete-math/discrete-math.md
Comment (CBID2) [inline content/discrete-math/discrete-math.md:15]: [Typo]
```suggestion
- **Efficiency:** Discrete mathematics enables the development of efficient algorithms and data structures, optimizing computational processes and reducing time complexity. This is crucial for handling large-scale data and solving complex problems in computer science and other domains.
```

File: content/discrete-math/discrete-math.md
Comment (CBID2) [inline content/discrete-math/discrete-math.md:16]: [Typo]
```suggestion
- **Problem-solving:** Discrete mathematics equips individuals with problem-solving skills and logical reasoning. It enables the analysis and process of breaking down complex problems into smaller, more manageable components, which in turn facilitates the development of effective solutions.
```

File: content/discrete-math/discrete-math.md
Comment (CBID2) [inline content/discrete-math/discrete-math.md:20]:
```suggestion
- **Logic and Reasoning:** Discrete mathematics entails using propositional and predicate logic, which are fundamental in computer science and critical for designing and verifying the correctness of software and hardware systems.
```

File: content/discrete-math/discrete-math.md
Comment (CBID2) [inline content/discrete-math/discrete-math.md:17]: [Clarity]
```suggestion
- **Algorithm Design:** Discrete mathematics include techniques such as graph theory, combinatorics, and optimization, which are essential for developing algorithms that efficiently solve various computational problems.
```

File: content/discrete-math/discrete-math.md
Comment (CBID2) [inline content/discrete-math/discrete-math.md:19]: [Clarity]
```suggestion
- **Cryptography and Security:** In Discrete mathematics, concepts like number theory, modular arithmetic, and encryption algorithms form the basis of cryptographic systems used for secure transactions, data privacy, and information security.
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi, @alimalim77! :) I pointed out a few proofreading errors in my inline comment. In the future, consider running your entries through [Grammarly](https://www.grammarly.com/grammar-check) or your word-processing software's spellcheck feature before making them into a pull request. Other than that, you're off to a great start! :)

**Comment by alimalim77** [comment]
Thanks @CBID2 for pointing out the errors. In keep this mind when creating a PR next time. Please approve this one or let me know of changes if any.

**Comment by CBID2** [comment]
> Thanks @CBID2 for pointing out the errors. In keep this mind when creating a PR next time. Please approve this one or let me know of changes if any.

I'm not a maintainer @alimalim77. Just another contributor who's here to help :)

**Review by CBID2 — COMMENTED** [review]
Looks good

**Comment by CBID2** [comment]
Hey @SSwiniarski and @KTom101. This entry is review ready

**Comment by SSwiniarski** [comment]
@alimalim77 Thanks for resubmitting, but this PR has the same issue as the last one. You need to reverse the edits to the `yarn.lock` file. Did you create a fresh branch off of main for this PR?

**Comment by alimalim77** [comment]
Yes I did create a fresh branch. I removed the yarn lock file while pushing. Is there anything I need to do for that? @SSwiniarski

**Comment by SSwiniarski** [comment]
> The problem seems to be this commit: [4a4d71f](https://github.com/Codecademy/docs/pull/2585/commits/4a4d71fcf8ab52a295ccee4e0a827c14fb1e0c10). The way you're running the formatting is disrupting the `yarn.lock` file. We can try and revert this commit and run `npm run format` on the two files. I can try doing that for you if you're uncomfortable with it.

@alimalim77 We crossed streams, I was editing my last reply. The problem seems to be this commit: [4a4d71f](https://github.com/Codecademy/docs/pull/2585/commits/4a4d71fcf8ab52a295ccee4e0a827c14fb1e0c10). The way you're running the formatting is disrupting the yarn.lock file. We can try and revert this commit and run npm run format on the two files. I can try doing that for you if you're uncomfortable with it.

**Comment by alimalim77** [comment]
I tried format all mentioned in Contributing Docs and that must have caused this I guess. Please do if possible @SSwiniarski

**Comment by SSwiniarski** [comment]
@alimalim77 sometimes `format all` can do this. you can avoid it in future by running it just on the files you're editing.

**Comment by SSwiniarski** [comment]
I wasn't able to revert the commit, but I was able to replace the modified `yarn.lock` with a fresh copy from `main`. I'll get rid of the invalid tag and do one more run through the entry.

File: content/discrete-math/concepts/proofs/proofs.md
Comment (SSwiniarski) [inline content/discrete-math/concepts/proofs/proofs.md:3]:
```suggestion
Description: 'Proofs are a concept in discrete mathematics used to verify the truthfulness and validity of statements using logical reasoning'
```

File: content/discrete-math/concepts/proofs/proofs.md
Comment (SSwiniarski) [inline content/discrete-math/concepts/proofs/proofs.md:17]:
```suggestion
**Proofs** are a concept in discrete mathematics used to verify the truthfulness and validity of statements. They use logical reasoning to establish the certainty of mathematical claims, playing a crucial role in advancing the field of mathematics. Proofs are also helpful in solving the problem partially or completely.
```

File: content/discrete-math/discrete-math.md
Comment (SSwiniarski) [inline content/discrete-math/discrete-math.md:3]:
```suggestion
Description: 'Discrete math is a fundamental branch of mathematics that provides the foundation for various concepts and techniques used in computer science and problem-solving.'
```

File: content/discrete-math/discrete-math.md
Comment (SSwiniarski) [inline content/discrete-math/discrete-math.md:13]:
```suggestion
Discrete mathematics offers the following advantages to the user:
```

**Review by SSwiniarski — APPROVED** [review]
@alimalim77 I have a few minor comments around capitalization, otherwise it looks good for a second review.

**Comment by alimalim77** [comment]
Made the changes. Please approve @SSwiniarski

**Review by SSwiniarski — APPROVED** [review]

**Review by CBID2 — APPROVED** [review]
LGTM

**Comment by alimalim77** [comment]
@SSwiniarski Is it okay or do I need to make any changes to it?

**Comment by SSwiniarski** [comment]
@alimalim77 It's fine. We're just waiting for another maintainer to do a second review.

File: content/discrete-math/discrete-math.md
Comment (caupolicandiaz) [inline content/discrete-math/discrete-math.md:4]: This should have subjects and tags instead of this link

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@alimalim77 I've completed my initial review. Please update the entry, and tag me when the changes are done. Thanks

File: content/discrete-math/discrete-math.md
Comment (alimalim77) [reply content/discrete-math/discrete-math.md:4]: The subjects are mentioned in catalog content right. I referred to other docs and they have similar structure.

Comment (alimalim77) [reply content/discrete-math/discrete-math.md]: I have added subjects and tags to it.

Comment (SSwiniarski) [reply content/discrete-math/discrete-math.md]: @caupolicandiaz, that's wrong because this is a topic page. It needs to follow [this template](https://github.com/Codecademy/docs/blob/main/documentation/topic-entry-template.md).

**Comment by SSwiniarski** [comment]
@alimalim77 can you undo the edit you made to the Discrete Math topic page, you had it right to start with.

**Comment by alimalim77** [comment]
@SSwiniarski I have undo the changes as requested

**Review by caupolicandiaz — APPROVED** [review]
@alimalim77 All set, looks good to go.

File: content/discrete-math/discrete-math.md
Comment (caupolicandiaz) [reply content/discrete-math/discrete-math.md:4]: My mistake

**Comment by KTom101** [comment]
@alimalim77 Thank you for your contributions:

Discrete Math: https://www.codecademy.com/resources/docs/discrete-math
Proofs: https://www.codecademy.com/resources/docs/discrete-math/proofs

You're currently listed as an anonymous user:

Screenshot 2023-07-03 at 11 58 14 PM

If you want to remain anonymous, there is nothing more you need to do. If you wish to be listed as a contributor, please follow the instructions [[in this forum discussion](https://discuss.codecademy.com/t/my-contribution-is-live-but-im-not-being-shown-as-a-contributor-why-is-this/758036)](https://discuss.codecademy.com/t/my-contribution-is-live-but-im-not-being-shown-as-a-contributor-why-is-this/758036).

cc @yangc95

**Pull request merged**
